cmd/tools/ldpreload-label-injector/inject.go

```diff
@@ -0,0 +1,425 @@
+				"  value: \"\"",
+				"- name: VCL_APP_SCOPE_LOCAL",
+				"  value: \"\"",
+			},
+			eol)
+	}
+
+	// proxy used
+	return insertLinesConditioned(
+		document,
+		[]string{"spec:", "template:", "spec:", "containers:", minusIndentationCharacter, "env:"},
+		isProxyContainer(params.proxyName),
+		[]string{ // proxy container settings
+			"# ldpreload-related env vars",
+			"- name: VCL_APP_SCOPE_GLOBAL",
+			"  value: \"\"",
+			"- name: VCL_APP_SCOPE_LOCAL",
+			"  value: \"\"",
+			"- name: VCL_APP_PROXY_TRANSPORT_TCP",
+			"  value: \"\"",
+		},
+		[]string{ // non-proxy container settings
+			"# ldpreload-related env vars",
+			"- name: VCL_APP_SCOPE_LOCAL",
+			"  value: \"\"",
+		},
+		eol,
+	)
+}
+
+// isProxyContainer checks if we are inside proxy image or not. This impl is tightly relying on path in insertAppScope() method.
+func isProxyContainer(proxyName string) conditionFunc {
+	return func(info traversingInfo) bool {
+		if len(info.unresolvedPath) > 1 {
+			return false // should not happen, because it would mean that there are no containers at all
+		}
+
+		imageBlockStr := info.blockStr(len(info.blocks) - (2 - len(info.unresolvedPath)))
+		return regexp.MustCompile("name: *"+proxyName+" *"+info.eol).
+			FindStringIndex(imageBlockStr) != nil
+	}
+}
+
+// insertDebug inserts ldpreload label that enables ldpreload debugging
+func insertDebug(document string, eol string) string {
+	return insertLines(
+		document,
+		[]string{"spec:", "template:", "spec:", "containers:", minusIndentationCharacter, "env:"},
+		[]string{
+			"# enable verbose VCL debugs, do not use for production",
+			"- name: VCL_DEBUG",
+			"  value: \"3\"",
+		},
+		eol)
+}
+
+// insertLines inserts <insertLines> to yaml <document> at position defined by <path>.
+// Yaml document is a block structured document. <path> is used to navigate from root block of document to the most
+// inner block where the lines should be inserted.
+func insertLines(document string, path []string, insertLines []string, eol string) string {
+	return insertLinesConditioned(document, path, func(traversingInfo) bool { return true }, insertLines, []string{}, eol)
+}
+
+type conditionFunc func(traversingInfo) bool
+
+// insertLinesConditioned is extended version of insertLines function and it can choose between 2 possible texts to insert based on condition (conditionFunc)
+func insertLinesConditioned(document string, path []string, condition conditionFunc, insertLines []string, elseInsertLines []string, eol string) string {
+	// visit places where text need to be inserted and by using information information collected by travelling (traversing info)
+	// build insertion information (where and what should be inserted)
+	// When text for insertion is constructed, it is not inserted right away because that would invalidate some
+	// indexes (navigation in document is index based, block are remembered as <startindex, endindex>)
+	var insertions = &[]insertion{}
```

Comment:
I believe there is no need to use pointer to slice

Reply:
Yes, you are right -> i fix that in another pull request...